objects/obj_player/Create_0.gml

```diff
@@ -119,6 +120,24 @@ state_free = function() {
 		speed = 0; //stop player movement while bouncing
 	}
 	
+	//check for collision with on off platform	
+	if (place_meeting(x,y+vspeed,obj_temp_platform_on_off) and !place_meeting(x,y-1,obj_temp_platform_on_off)) and vspeed > 0 and platform_on {
+		while !(place_meeting(x,y+sign(vspeed),obj_temp_platform_on_off)) {
+			y += sign(vspeed);
+		}
+		state = state_bouncing;
+		speed = 0; //stop player movement while bouncing
+	}
+	
+	//check for collision with off on platform	
```

Comment:
Is this part necessary? Seems like you could remove the part with "and platform_on" and they would both function the same